src/EditorFeatures/Core/Shared/Preview/PreviewSolutionCrawlerRegistrationService.cs

```diff
@@ -101,7 +101,12 @@ private async Task AnalyzeAsync()
                 }
             }
 
-            public async void Unregister(Workspace workspace, bool blockingShutdown = false)
+            public void Unregister(Workspace workspace, bool blockingShutdown = false)
+            {
+                _ = UnregisterAsync(workspace, blockingShutdown);
+            }
+
+            private async Task UnregisterAsync(Workspace workspace, bool blockingShutdown)
             {
                 Contract.ThrowIfFalse(workspace == _workspace);
                 _source.Cancel();
```

Comment:
what happens if awaitin _analyzeTask below throws?

Reply:
➡️ Prior to this pull request, the application would crash. With this pull request, the exception will be ignored. With the changes for VSTHRD110, I am guessing we will opt for the approach of NFW.